crawler/base_crawler.py

```diff
@@ -32,20 +32,113 @@ def __str__(self):
         return '\n'.join(str(feature) for feature in self.data)
 
 
-class BaseCrawler:
+class Crawler:
 
-    def __init__(self, features=['os', 'cpu'],
+    def __init__(self,
+                 emitters=None,
+                 frequency=-1,
+                 features=['os', 'cpu'],
                  plugin_places=['plugins'],
                  options={}):
+        """
+        Store and check the types of the arguments.
+
+        :param emitters: EmittersManager that holds the list of Emitters.
+        If it is None, then no emit is done.
+        :param frequency: Sleep seconds between iterations
+        """
+        self.iter_count = 0
+        self.frequency = frequency
+        self.next_iteration_time = None
+        self.emitters = emitters
         self.features = features
         self.plugin_places = plugin_places
         self.options = options
 
     def crawl(self, ignore_plugin_exception=True):
         """
-        Crawl to get a snapshot frame of all systems.
+        Crawl to get a list of snapshot frames for all systems.
 
         :param ignore_plugin_exception: ignore exceptions raised on a plugin
         :return: a list generator of Frame objects
         """
-        pass
+        raise NotImplementedError('crawl method implementation is missing.')
+
+    def polling_crawl(self, timeout, ignore_plugin_exception=True):
```

Comment:
The comment from the iterate function can be added here: "If polling is not implemented in subclass, this is a sleep(timeout)